src/internet_identity/src/vc_mvp.rs

```diff
@@ -0,0 +1,247 @@
+        rp_id_alias_jwt,
+        issuer_id_alias_jwt,
+    }
+}
+
+fn id_alias_jwt_and_msg_hash(
+    alias_tuple: &AliasTuple,
+    canister_sig_pk: &CanisterSigPublicKey,
+) -> (String, Hash) {
+    let credential_jwt = prepare_id_alias_jwt(alias_tuple);
+    let signing_input =
+        vc_signing_input(&credential_jwt, canister_sig_pk).expect("failed getting signing_input");
+    let msg_hash = vc_signing_input_hash(&signing_input);
+    (credential_jwt, msg_hash)
+}
+
+pub fn get_id_alias(
+    identity_number: IdentityNumber,
+    dapps: InvolvedDapps,
+    rp_id_alias_jwt: &str,
+    issuer_id_alias_jwt: &str,
+) -> GetIdAliasResponse {
+    check_frontend_length(&dapps.relying_party);
+    check_frontend_length(&dapps.issuer);
+
+    state::assets_and_signatures(|cert_assets, sigs| {
+        let id_alias_principal = get_id_alias_principal(identity_number, &dapps);
+        let seed = calculate_id_alias_seed(identity_number, &dapps);
+        let id_rp = delegation::get_principal(identity_number, dapps.relying_party.clone());
+        let id_issuer = delegation::get_principal(identity_number, dapps.issuer.clone());
+        let canister_id = ic_cdk::id();
+        let canister_sig_pk = CanisterSigPublicKey::new(canister_id, seed.to_vec());
+
+        let signing_input = vc_signing_input(rp_id_alias_jwt, &canister_sig_pk)
+            .expect("failed getting signing_input");
+        let msg_hash = vc_signing_input_hash(&signing_input);
+        let maybe_sig = get_signature(cert_assets, sigs, seed, msg_hash);
+        let rp_sig = if let Some(sig) = maybe_sig {
+            sig
+        } else {
+            return GetIdAliasResponse::NoSuchCredentials("rp_sig not found".to_string());
+        };
+        let rp_jws = vc_jwt_to_jws(rp_id_alias_jwt, &canister_sig_pk, &rp_sig)
+            .expect("failed constructing JWS");
+
+        let signing_input = vc_signing_input(issuer_id_alias_jwt, &canister_sig_pk)
+            .expect("failed getting signing_input");
+        let msg_hash = vc_signing_input_hash(&signing_input);
+        let maybe_sig = get_signature(cert_assets, sigs, seed, msg_hash);
+        let issuer_sig = if let Some(sig) = maybe_sig {
+            sig
+        } else {
+            return GetIdAliasResponse::NoSuchCredentials("issuer_sig not found".to_string());
+        };
+        let issuer_jws = vc_jwt_to_jws(issuer_id_alias_jwt, &canister_sig_pk, &issuer_sig)
+            .expect("failed constructing JWS");
+
+        GetIdAliasResponse::Ok(IdAliasCredentials {
+            rp_id_alias_credential: SignedIdAlias {
+                id_alias: id_alias_principal,
+                id_dapp: id_rp,
+                credential_jws: rp_jws,
+            },
+            issuer_id_alias_credential: SignedIdAlias {
+                id_alias: id_alias_principal,
+                id_dapp: id_issuer,
+                credential_jws: issuer_jws,
+            },
+        })
+    })
+}
+
+fn get_signature(
```

Comment:
This is a lot of duplication from the same method in `delegation.rs`.
What about only having it once and make the one from `delegation.rs` also take a `msg_hash` instead of the `(pk, expiration)` inputs to the delegation?